### ISSUE TITLE
PSR12.Files.FileHeader.SpacingAfterBlock: false positive when `use` is used for class and function

### ISSUE THREAD

**Issue by VladyslavSikailo** [squizlabs/PHP_CodeSniffer#3935]
## Describe the bug
PSR12.Files.FileHeader.SpacingAfterBlock check is false positive when `use` is used for class and function.

### Code sample
```php
<?php

namespace Foo;

use Countable;
use function str_replace;

class Test
{
}
```

### To reproduce
1. Create a file called `Test.php` with the code sample above
2. Run `phpcs -s --standard=PSR12 --no-cache Test.php`
3. See the error message displayed
```
-------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-------------------------------------------------------------------------------------------------------------------
 5 | ERROR | [x] Header blocks must be separated by a single blank line
   |       |     (PSR12.Files.FileHeader.SpacingAfterBlock)
-------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-------------------------------------------------------------------------------------------------------------------
```

## Expected behavior
Checking should be passed

## Versions (please complete the following information)

| | |
|-|-|
| Operating System | MacOS 14.0
| PHP version | 8.2.6
| PHP_CodeSniffer version | master
| Standard | PSR12
| Install type | Composer (local)

## Please confirm:

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [x] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by jrfnl** [comment]
@VladyslavSikailo Not a false positive. [PSR12 demands a blank line between different type of `use` statements](https://www.php-fig.org/psr/psr-12/#3-declare-statements-namespace-and-import-statements), so there should be a blank line between the last class `use` statement and the first function `use` statement.

P.S.: for the future, please open new issues in https://github.com/PHPCSStandards/PHP_CodeSniffer - also see #3932
